crates/ruff_db/src/system/os.rs

```diff
@@ -336,6 +350,70 @@ fn not_found() -> std::io::Error {
     std::io::Error::new(std::io::ErrorKind::NotFound, "No such file or directory")
 }
 
+#[cfg(feature = "testing")]
+pub(super) mod testing {
+    use std::cell::RefCell;
+
+    use crate::system::SystemPathBuf;
+
+    thread_local! {
```

Comment:
Just to make sure I understand. We can't add something like a `#cfg(test)`-level `set_user_configuration_directory_override(…)` function to `OsSystem`, because we use the same `OsSystem` from multiple tests (and therefore, multiple threads)?

Reply:
Hmm, no that was just me being dumb... I should do that instead because it fixes the single-thread limitation. It still wont work in CLI tests but that's because those tests spawn a new process

Reply:
This turned out a bit annoying because it is now required to pass around the `OsSystem` in the `file_watching` tests but it's way more robust. Thanks for suggesting this. I completely missed that we have an instance where we could store the state on.